portal/portal-service-impl/impl/src/java/org/sakaiproject/portal/service/PortalServiceImpl.java

```diff
@@ -950,12 +950,11 @@ public void savePinnedSites(List<String> siteIds) {
 
 		siteIds.removeAll(currentPinned);
 
-        for (int i = getPinnedSites().size(); i < siteIds.size(); i++) {
-
-            String siteId = siteIds.get(i);
-
+		int start = getPinnedSites().size();
+		for (int i = 0; i < siteIds.size(); i++) {
+			String siteId = siteIds.get(i);
 			PinnedSite pin = pinnedSiteRepository.findByUserIdAndSiteId(userId, siteId).orElseGet(() -> new PinnedSite(userId, siteId));
-			pin.setPosition(i);
+			pin.setPosition(i + start);
```

Comment:
It would be nice to understand why the order starts after the current pinned list, are these new additions?

Reply:
Yes, the sites processed by this portion of code (as asserted prior to SAK-48948) places newly pinned sites at the bottom of the Pinned sites listed in the site nav. 

I didn't insert a clarifying comment in the code about the intended behavior because this logic pre-existed SAK-48948 where no such comment existed. In the above section of code, the fix for SAK-48948 merged into trunk/24.x had introduced the regression described in SAK-49332.